naptime-graphql/src/main/scala/org/coursera/naptime/ari/graphql/schema/FieldBuilder.scala

```diff
@@ -0,0 +1,167 @@
+package org.coursera.naptime.ari.graphql.schema
+
+import com.linkedin.data.DataMap
+import com.linkedin.data.schema.ArrayDataSchema
+import com.linkedin.data.schema.BooleanDataSchema
+import com.linkedin.data.schema.BytesDataSchema
+import com.linkedin.data.schema.DoubleDataSchema
+import com.linkedin.data.schema.EnumDataSchema
+import com.linkedin.data.schema.FloatDataSchema
+import com.linkedin.data.schema.IntegerDataSchema
+import com.linkedin.data.schema.LongDataSchema
+import com.linkedin.data.schema.MapDataSchema
+import com.linkedin.data.schema.RecordDataSchema
+import com.linkedin.data.schema.RecordDataSchema.{Field => RecordDataSchemaField}
+import com.linkedin.data.schema.StringDataSchema
+import com.linkedin.data.schema.TyperefDataSchema
+import com.linkedin.data.schema.UnionDataSchema
+import org.coursera.naptime.ari.graphql.SangriaGraphQlContext
+import org.coursera.naptime.ari.graphql.types.NaptimeTypes
+import sangria.schema.BooleanType
+import sangria.schema.Context
+import sangria.schema.Field
+import sangria.schema.FloatType
+import sangria.schema.IntType
+import sangria.schema.ListType
+import sangria.schema.LongType
+import sangria.schema.OptionType
+import sangria.schema.StringType
+import sangria.schema.Value
+
+import scala.collection.JavaConverters._
+
+
+object FieldBuilder {
+
+  type ResolverType = Context[SangriaGraphQlContext, DataMap] => Value[SangriaGraphQlContext, Any]
+
+  def buildField(
+      schemaMetadata: SchemaMetadata,
+      field: RecordDataSchemaField,
+      namespace: Option[String],
+      fieldNameOverride: Option[String] = None): Field[SangriaGraphQlContext, DataMap] = {
+    type ResolverType = Context[SangriaGraphQlContext, DataMap] => Value[SangriaGraphQlContext, Any]
+
+    val relatedResourceOption = field.getProperties.asScala.get("related").map(_.toString)
+
+    val fieldName = fieldNameOverride.getOrElse(field.getName)
+
+    val fieldDoc = if (field.getDoc.isEmpty) None else Some(field.getDoc)
```

Comment:
I'm not sure I understand. `Field.getDoc` will return the doc from the record type, i.e.
```
record Course {
  /**
   * This field denotes the main instructors for the course
   */
  primaryInstructors: InstructorList

 /**
   * This field denotes the backup instructors for the course
   */
  backupInstructors: InstructorList
}```

We'd want to differentiate in the doc here between primaryInstructors and backupInstructors -- if we went to the documentation on InstructorList, it would be the same for both.

In GraphiQl, people can get the documentation for the field, and then on the type itself, see the documentation. (i.e. documentation on both primaryInstructors and InstructorList would be available).